namespaces/exec.go

```diff
@@ -90,11 +94,46 @@ func Exec(container *libcontainer.Config, stdin io.Reader, stdout, stderr io.Wri
 		CgroupPaths:   cgroupPaths,
 	}
 
+	log.Println("Saving state.")
 	if err := libcontainer.SaveState(dataPath, state); err != nil {
 		return terminate(err)
 	}
 	defer libcontainer.DeleteState(dataPath)
 
+	// Start the setup process to setup the init process
+	if container.Namespaces.Contains(libcontainer.NEWUSER) {
+		log.Println("Starting setup")
+		setupCmd := setupCommand(container, console, dataPath, os.Args[0])
+		setupOut, _ := setupCmd.StderrPipe()
```

Comment:
Yes, that should work. I went back and forth while working through it. Added to my list for today.